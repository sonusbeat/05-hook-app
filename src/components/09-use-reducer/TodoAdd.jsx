import useForm from '../../hooks/useForm';

const TodoAdd = ({ HandleAddTodo }) => {
  const [{ description }, handleInputChange, reset] = useForm({ description: "" });

  const HandleSubmit = ( event ) => {
    event.preventDefault();

    // Validation
    if (description.trim().length <= 2) return;

    const newTodo = {
      id: new Date().getTime(),
      description: description,
      done: false
    };

    HandleAddTodo(newTodo);

    reset();
  };

  return (
    <div className="mb-5">
      <form onSubmit={ HandleSubmit }>
        <div className="form-group">
          <input
            name="description"
            type="text"
            className="form-control"
            autoComplete="off"
            onChange={ handleInputChange }
            value={ description }
          />
        </div>
        <button type="submit" className="btn btn-primary btn-block">
          Agregar
        </button>
      </form>
    </div>
  );
};

export default TodoAdd;
