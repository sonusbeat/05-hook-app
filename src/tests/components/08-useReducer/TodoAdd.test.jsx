import TodoAdd from '../../../components/09-use-reducer/TodoAdd';
import { shallow } from 'enzyme';
// import demoTodos from '../../fixtures/demotTodos';

describe('Pruebas en <TodoAdd />', () => {

  const HandleAddTodo = jest.fn();

  const wrapper = shallow(
    <TodoAdd
      HandleAddTodo={ HandleAddTodo }
    />
  );

  test("Debe de coincidir con el snapshot", () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test("No debe de llamar HandleAddTodo", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");

    formSubmit({ preventDefault(){} });

    expect( HandleAddTodo ).toHaveBeenCalledTimes(0);
  });

  test("Debe de llamar la funciÃ³n HandleAddTodo", () => {
    const value = "Learn React";

    wrapper.find("input").simulate("change", {
      target: {
        value,
        name: "description"
      }
    });

    const formSubmit = wrapper.find("form").prop("onSubmit");

    formSubmit({ preventDefault(){} });

    expect( HandleAddTodo ).toHaveBeenCalledTimes(1);
    expect( HandleAddTodo ).toHaveBeenCalledWith( expect.any(Object) );
    expect( HandleAddTodo ).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: value,
      done: false,
    });

    expect(wrapper.find("input").prop("value")).toBe("");

  });


});
