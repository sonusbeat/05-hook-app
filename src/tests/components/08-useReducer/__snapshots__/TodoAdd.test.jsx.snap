// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> Debe de coincidir con el snapshot 1`] = `
<div
  className="mb-5"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        autoComplete="off"
        className="form-control"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</div>
`;
